### PR TITLE
[MM-43787] Replace hardcoded PGP key with embedded one

### PR DESCRIPTION
**Pull request by noxer**
#### Summary
This PR replaces the hard-coded PGP public key with an embedded one for `services/upgrader`. It also fixes a number of issues in the upgrader.
- Since the `Content-Length` is reported by the server it also may be -1 (unknown), 0, or some random number, while the server sends more bytes. We now do a small sanity check before using it to calculate the percentage.
- Prevent the unchecked concurrent writes/reads of `upgradePercentage` and `upgradeError`.

#### Ticket Link
[MM-43787](https://mattermost.atlassian.net/browse/MM-43787)

#### Release Note
```release-note
NONE
```

### CODE REVIEW

File: services/upgrader/upgrader_linux.go [hunk 1]
```diff
@@ -27,48 +29,65 @@ import (
...
-maBevuNymlFbODTciyE9Q7mJHkaamGKTXaa3Enlcf16oSoemawSBJuspaS0sZOW8
-dgi5l3V5YvfFvSk45axiZbnGYfN81G5mkSGAENSGSKVtMA==
-=kkvg
------END PGP PUBLIC KEY BLOCK-----`
-
-var upgradePercentage int64
-var upgradeError error
-var upgrading int32
+//go:embed pubkey.gpg
+var mattermostBuildPublicKeys []byte
```
Comment (jupenur) [inline services/upgrader/upgrader_linux.go:33]: Nit: Why the plural? We should never have more than one key here, right?

**Review by jupenur — APPROVED** [review]

File: services/upgrader/upgrader_linux.go (hunk 1 above)
Comment (noxer) [reply services/upgrader/upgrader_linux.go:33]: There can be multiple keys in the file. This enables us to add the next key as well. We need to test this properly, though.

Comment (DSchalla) [reply services/upgrader/upgrader_linux.go:33]: Sorry folks, I didn't share prior context with @jupenur prior to reassigning. I linked to the thread where this was discussed.

**Review by spirosoik — APPROVED** [review]

File: services/upgrader/upgrader_linux.go [hunk 2]
```diff
@@ -27,48 +29,65 @@ import (
...
+
+func setUpgradePercentage(to int64) {
+	atomic.StoreInt64(&upgradePercentage, to)
+}
+
+func getUpgradeError() error {
+	m.Lock()
+	defer m.Unlock()
+
+	return upgradeError
```
Comment (agnivade) [inline services/upgrader/upgrader_linux.go:81]: Maybe we can use [atomic.Value](https://pkg.go.dev/sync/atomic#Value) for this?

Comment (noxer) [reply services/upgrader/upgrader_linux.go:81]: Unfortunately that's not possible. `atomic.Value` does not support storing `nil` [1] and after the first `Store` is fixed to this particular type [2], making it impossible to store different underlying error types (like we do).

[1] https://cs.opensource.google/go/go/+/refs/tags/go1.18.3:src/sync/atomic/value.go;l=48
[2] https://cs.opensource.google/go/go/+/refs/tags/go1.18.3:src/sync/atomic/value.go;l=77

Comment (agnivade) [reply services/upgrader/upgrader_linux.go:81]: Ah interesting. Thanks!

**Review by agnivade — APPROVED** [review]

**Pull request merged**